state/protocol/badger/mutator.go

```diff
@@ -877,6 +877,7 @@ func (m *FollowerState) epochTransitionMetricsAndEventsOnBlockFinalized(block *f
 // Suppose an EpochSetup service event is emitted during execution of block A. C seals A, therefore
 // we apply the metrics/events when C is finalized. The first block of the EpochSetup
 // phase is block C.
+// TODO: this should read data from the KVStore/ProtocolStateEntry for the block rather than parsing service events
```

Comment:
lets capture this in an issue. I am worried that if we forget that, there might be an unnecessary pager duty alert going out

Reply:
👍  It's captured in https://github.com/onflow/flow-go/issues/5666.